app/src/org/commcare/android/database/SqlStorage.java

```
@@ -252,7 +273,11 @@ public int add(Externalizable e) throws StorageFullException {
      * @see org.javarosa.core.services.storage.IStorageUtility#close()
      */
     public void close() {
-        helper.getHandle().close();
+        try {
+            helper.getHandle().close();
+        } catch (SessionUnavailableException e) {
+            // The db isn't available so don't worry about closing it.
+        }
```

Swallow the exception in this case.